### PR TITLE
enable dark theme

### LINKED ISSUE #3055: docs: toggle icon to switch to dark mode
**Is your feature request related to a problem? Please describe.**
mkdocs support dark mode and I would like to have it enabled on the documentation site. 

**Describe the solution you'd like**
A toggle icon to switch between dark and light mode would be great. 

**Describe alternatives you've considered**
mkdocs docs itself have it enabled and I think this can be enabled by referring this.
https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#color-palette-toggle

**Additional context**

### PR DESCRIPTION
**Pull request by andylim0221**
## Problem Statement

Enable dark theme.

## Related Issue

Fixes #3055 

## Proposed Changes

Enable theme toggle in mkdocs config file.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: hack/api-docs/mkdocs.yml [hunk 1]
```diff
@@ -7,11 +7,17 @@ edit_uri: ./edit/main/docs/
 remote_branch: gh-pages
 theme:
   name: material
-  features:
```
Comment (Skarlso) [inline hack/api-docs/mkdocs.yml:10]: Nice but why delete the features section?

Comment (Skarlso) [reply hack/api-docs/mkdocs.yml:10]: Thanks for the fix.

**Comment by moolen** [comment]
I think we need to review all the images and rework some of them. Some diagrams unfortunately are not readable.

mkdocs-material allows us to use a switch: https://squidfunk.github.io/mkdocs-material/reference/images/#light-and-dark-mode However, then we would have to maintain different images for each palette. 

I've attached a couple of screenshots of unreadable images, these are not all.
Screenshot from 2024-01-22 20-39-58
Screenshot from 2024-01-22 20-38-55
Screenshot from 2024-01-22 20-38-45

**Comment by Skarlso** [comment]
I feared something like that would happen. I literally was just about to see how they render :D Nice timing. :)

**Comment by andylim0221** [comment]
However updating the affected or all images by adding solid background color? That way, we don't need to maintain for each palette.

**Comment by moolen** [comment]
IMHO: 

The solid background color will look bad due to the high contrast (white bg on dark mode). However, as a compromise that would work out. We don't have _that_ many diagrams. Though for the important pages (landing page, introduction) we really should create new diagrams that work well in dark mode.

**Comment by andylim0221** [comment]
@moolen If that's the case, probably switching image based on theme would be the best choice in long run, as the one you suggested: https://squidfunk.github.io/mkdocs-material/reference/images/#light-and-dark-mode

as of now, we just need to create the copies for the images with transparent background with dark theme background added. If so, how should I proceed to modify the images?

**Comment by Skarlso** [comment]
@moolen ? If the images are unreadable that's bad we can't just have them as light. But if they are okay for now I would suggest to just go ahead with adding dark mode than later we can change the images.

**Comment by moolen** [comment]
> @moolen ? If the images are unreadable that's bad we can't just have them as light. But if they are okay for now I would suggest to just go ahead with adding dark mode than later we can change the images.

If we were to add dark mode it would be the default for some users (80%+ of mobile users)  - i wouldn't want to release it in a unrefined way. I guess as long as the important pages (landing page, introduction, API...) are readable then this is good enough

**Review by Skarlso — APPROVED** [review]

**Pull request merged**